pkg/kncloudevents/http_client_new.go

```diff
@@ -29,42 +30,63 @@ import (
 )
 
 const (
-	defaultRetryWaitMin = 1 * time.Second
-	defaultRetryWaitMax = 30 * time.Second
+	defaultRetryWaitMin    = 1 * time.Second
+	defaultRetryWaitMax    = 30 * time.Second
+	defaultClientsTTL      = 30 * time.Minute
+	defaultRecheckInterval = 5 * time.Minute
 )
 
 var (
 	clients clientsHolder
 )
 
 type clientsHolder struct {
-	mu             sync.Mutex
-	clients        map[string]*nethttp.Client
-	connectionArgs *ConnectionArgs
+	clientsMu       sync.Mutex
+	clients         map[string]*clientsMapEntry
+	timerMu         sync.Mutex
+	connectionArgs  *ConnectionArgs
+	ttl             time.Duration
+	recheckInterval time.Duration
+	cancelCleanup   context.CancelFunc
+}
+
+type clientsMapEntry struct {
+	client *nethttp.Client
+	expiry int64
```

Comment:
Instead of setting the expiry, I would set the last-access-date. This allows you to take changes on the TTL directly into account.
And are there reasons for using an `int64` type, instead of using `time.Time`?

Reply:
Accessing the client doesn't mean I'm not using it though, right?

Reply:
That's true. But wouldn't it be the same with using `expiry`? And through receiving a client should be done only through the private `getClientForAddressable` function, I would see this mostly under our control anyhow (if I understood your point correctly).

Reply:
@creydr I've used last-access-date, and switched to use `time.Time` instead of `int64`